CS483/CS483/Kartaclysm/Components/ComponentTrack.cpp

```diff
@@ -0,0 +1,234 @@
+#include "ComponentTrack.h"
+
+#include "KeyboardInputBuffer.h"
+#include "EventManager.h"
+#include <iostream>
+
+namespace Kartaclysm
+{
+	ComponentTrack::ComponentTrack(HeatStroke::GameObject* p_pGameObject, const std::string& p_strTrackName)
+		:
+		Component(p_pGameObject),
+		m_strTrackName(p_strTrackName),
+		m_vTrackPieces()
+	{
+		std::function<void(const HeatStroke::Event*)>* func = new std::function<void(const HeatStroke::Event*)>(std::bind(&ComponentTrack::OnRacerTrackPieceCollision, this, std::placeholders::_1));
+		HeatStroke::EventManager::Instance()->AddListener("RacerTrackPieceUpdated", func);
+	}
+
+	ComponentTrack::~ComponentTrack()
+	{
+	}
+
+	HeatStroke::Component* ComponentTrack::CreateComponent(HeatStroke::GameObject* p_pGameObject, tinyxml2::XMLNode* p_pBaseNode, tinyxml2::XMLNode* p_pOverrideNode)
+	{
+		std::string strTrackName = "";
+		HeatStroke::EasyXML::GetRequiredStringAttribute(p_pBaseNode->FirstChildElement("Name"), "value", strTrackName);
+
+		return new ComponentTrack(p_pGameObject, strTrackName);
+	}
+
```

Comment:
I'd like to see this follow our standard format of parsing a base and override using a ParseNode method. This future proofs us if this feature gets re-implemented.